server/text_generation_server/models/causal_lm.py

```diff
@@ -630,6 +642,52 @@ def __init__(
             self.hb_profer_started = False
         self.step = 0
 
+    def setup_env(self):
```

Comment:
Why is this a member? The only thing it does on the object itself is setting is_quantizaton_enabled which we can check independently after setup. Once we move it outside we can call it before loading any modules so that the environment is set before loading habana_frameworks